employee-rostering-frontend/src/store/contract/contract.test.ts

```diff
@@ -0,0 +1,205 @@
+    onPost(`/tenant/${tenantId}/contract/add`, contractToAdd, contractWithUpdatedId);
+    await store.dispatch(contractOperations.addContract(contractToAdd));
+    expect(store.getActions()).toEqual([actions.addContract(contractWithUpdatedId)]);
+    expect(client.post).toHaveBeenCalledTimes(1);
+    expect(client.post).toHaveBeenCalledWith(`/tenant/${tenantId}/contract/add`, contractToAdd);
+
+    store.clearActions();
+    client.post.mockClear()
+
+    const contractToUpdate: Contract = {tenantId: tenantId,
+      name: "Contract 1",
+      id: 4,
+      version: 0,
+      maximumMinutesPerDay: null,
+      maximumMinutesPerWeek: null,
+      maximumMinutesPerMonth: null,
+      maximumMinutesPerYear: null
+    };
+    const contractWithUpdatedVersion: Contract = {...contractToUpdate, id: 4, version: 1};
+    onPost(`/tenant/${tenantId}/contract/update`, contractToUpdate, contractWithUpdatedVersion);
+    await store.dispatch(contractOperations.updateContract(contractToUpdate));
+    expect(store.getActions()).toEqual([actions.updateContract(contractWithUpdatedVersion)]);
+    expect(client.post).toHaveBeenCalledTimes(1);
+    expect(client.post).toHaveBeenCalledWith(`/tenant/${tenantId}/contract/update`, contractToUpdate);
+  });
+});
+
+describe('Contract reducers', () => {
+  const addedContract: Contract = {
+    tenantId: 0,
+    id: 4,
+    version: 0,
+    name: "Contract 4",
+    maximumMinutesPerDay: null,
+    maximumMinutesPerWeek: 1,
+    maximumMinutesPerMonth: 2,
+    maximumMinutesPerYear: 3
+  };
+  const updatedContract: Contract = {
+    tenantId: 0,
+    id: 1,
+    version: 0,
+    name: "Updated Contract 2",
+    maximumMinutesPerDay: 1,
+    maximumMinutesPerWeek: 2,
+    maximumMinutesPerMonth: 3,
+    maximumMinutesPerYear: 4
+  };
+  const deletedContract: Contract =       {
+    tenantId: 0,
+    id: 2,
+    version: 0,
+    name: "Contract 3",
+    maximumMinutesPerDay: 100,
+    maximumMinutesPerWeek: null,
+    maximumMinutesPerMonth: null,
+    maximumMinutesPerYear: 100
+  };
+  it('add contract', () => {
+    expect(
+      reducer(state.contractList, actions.addContract(addedContract))
+    ).toEqual({contractList: withElement(state.contractList.contractList, addedContract)})
+  });
+  it('remove contract', () => {
+    expect(
+      reducer(state.contractList, actions.removeContract(deletedContract)),
+    ).toEqual({contractList: withoutElement(state.contractList.contractList, deletedContract)})
+  });
+  it('update contract', () => {
+    expect(
+      reducer(state.contractList, actions.updateContract(updatedContract)),
+    ).toEqual({contractList: withUpdatedElement(state.contractList.contractList, updatedContract)})
+  });
+  it('refresh contract list', () => {
+    expect(
+      reducer(state.contractList, actions.refreshContractList([addedContract])),
+    ).toEqual({contractList: [addedContract]});
+  });
+});
+
+const state: AppState = {
```

Comment:
TypeScript compiler:
```
TS2741: Property 'contractList' is missing in type '{ tenantData: { currentTenantId: number; tenantList: never[]; }; spotList: { spotList: never[]; }; skillList: { skillList: { tenantId: number; id: number; version: number; name: string; }[]; }; }' but required in type 'AppState'.
```

Reply:
Did you do it after updating the import to  `store/rest/__mocks__/RestServiceClient`? I can reproduce the error only after updating the import to `store/rest/__mocks__/RestServiceClient`; no errors without updating the import.

Reply:
Seems you commented on the wrong file; `tenant.test` is the one with the issue.